### PR TITLE
Check for ICN in addition to PID

### PR DESCRIPTION
**Pull request by danlim715**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- When checking for users with missing PIDs, first check for presence of ICN, because if ICN doesn't exist, PID cannot exist either.
- Pension benefits team to own this change.

## Related issue(s)

- [*Link to ticket created in va.gov-team repo*](https://github.com/department-of-veterans-affairs/va.gov-team/issues/88635)

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/controllers/v0/in_progress_forms_controller.rb [hunk 1]
```diff
@@ -26,12 +26,13 @@ def update
 
       form.update!(form_data: params[:form_data] || params[:formData], metadata: params[:metadata])
 
+      veteran_icn = @current_user.icn
       if Flipper.enabled?(:intent_to_file_lighthouse_enabled, @current_user) && form.id_previously_changed? &&
          Lighthouse::CreateIntentToFileJob::ITF_FORMS.include?(form.form_id)
-        if @current_user.participant_id.blank?
+        if veteran_icn.present? && @current_user.participant_id.blank?
```
Comment (mjknight50) [inline app/controllers/v0/in_progress_forms_controller.rb:32]: Is this right?  Should it be !veteran_icn.present?

Comment (danlim715) [reply app/controllers/v0/in_progress_forms_controller.rb:32]: my understanding of this issue was that if ICN is missing, then PIDs will always be blanks, so no need to log those instances.  We should only be logging instances where ICN is present,  but PID is missing. Let me know if I've misunderstood something.

Comment (mjknight50) [reply app/controllers/v0/in_progress_forms_controller.rb:32]: OK, could be written like this?
`if !veteran_icn.present? || @current_user.participant_id.blank?`

**Review by mjknight50 — APPROVED** [review]
GTG

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**